### PR TITLE
feat: [SRM-13553]: implementing the scenario of clicking on Configurations tab

### PR DESCRIPTION
**Pull request by harshilbits**
### Summary
1. Implementing the scenario of clicking on Configurations tab and data not getting reflected in Edit scenario of health source.
2. Refactoring to generate correct dataSourcetype for NextgenHealth source.

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by douglasnfrazao — DISMISSED** [review]

File: src/modules/85-cv/pages/health-source/HealthSourceDrawer/component/customiseHealthSource/CustomiseHealthSource.utils.tsx [hunk 1]
```diff
@@ -57,9 +57,12 @@ export const LoadSourceByType = ({
 }): JSX.Element | null => {
   const isSplunkMetricEnabled = useFeatureFlag(FeatureFlag.CVNG_SPLUNK_METRICS)
   const isSumoLogicEnabled = useFeatureFlag(FeatureFlag.SRM_SUMO)
-  const healthSourceConfig = healthSourcesConfig[data?.product?.value]
+  const selectedProduct = data?.product?.value || data?.existingMetricDetails?.spec?.dataSourceType
```
Comment (praneshg239) [inline src/modules/85-cv/pages/health-source/HealthSourceDrawer/component/customiseHealthSource/CustomiseHealthSource.utils.tsx:60]: `dataSourceType` is only for prometheus right? it is added for `AWSPrometheus`

Comment (harshilbits) [reply src/modules/85-cv/pages/health-source/HealthSourceDrawer/component/customiseHealthSource/CustomiseHealthSource.utils.tsx:60]: Its also used in the new health source design .
Where type is given as "NextGenHealthSoure" and dataSourceType is given as the exact feature selected for eg. "SUMOLOGIC_CLOUD" or "SUMOLOGIC_METRICS".

**Comment by harshilbits** [comment]
retrigger jest

**Review by praneshg239 — APPROVED** [review]

File: src/modules/85-cv/pages/health-source/types.ts [hunk 2]
```diff
@@ -42,6 +42,8 @@ export enum HealthSourceTypes {
   Elk = 'ElasticSearch',
   AwsPrometheus = 'AwsPrometheus',
   SumoLogic = 'SumoLogic',
+  SumoLogicMetrics = 'SUMOLOGIC_METRICS',
```
Comment (deepesh-ui) [inline src/modules/85-cv/pages/health-source/types.ts:45]: should we be removing SumoLogic = 'SumoLogic',

Comment (harshilbits) [reply src/modules/85-cv/pages/health-source/types.ts:45]: Yes , that will be removed once Anshuman's new set of changes are merged.

**Comment by harshilbits** [comment]
Coverage for old files is not meeting .

- CustomiseHealthSource.tsx
- CustomiseHealthSource.utils.tsx

Tests have been added , for the changes we did in these files.

**Review by deepesh-ui — APPROVED** [review]

**Pull request merged**